scripts/create-package.sh

```diff
@@ -0,0 +1,243 @@
+    DESTDIR="$(realpath "$pkg_dir")" make -C contrib/icons/ install datadir=
+}
+
+pkg_macos()
+{
+    # Note, this script assumes macos builds have x86_64 and ARM64 subdirectories
+
+    # Print shared object dependencies
+    otool -L "${build_dir}/dosbox-arm64/dosbox"
+    python3 scripts/verify-macos-dylibs.py "${build_dir}/dosbox-arm64/dosbox"
+
+    # Create universal binary from both architectures
+    mkdir dosbox-universal
+    lipo dosbox-x86_64/dosbox dosbox-arm64/dosbox -create -output dosbox-universal/dosbox
+
+    # Generate icon
+    make -C contrib/icons/ dosbox-staging.icns
+
+    install -d   "${macos_dst_dir}/MacOS/"
+    install      dosbox-universal/dosbox           "${macos_dst_dir}/MacOS/"
+    install_file contrib/macos/Info.plist.template "${macos_dst_dir}/Info.plist"
+    install_file contrib/macos/PkgInfo             "${macos_dst_dir}/PkgInfo"
+    install_file contrib/icons/dosbox-staging.icns "${macos_dst_dir}/Resources/"
+
+    sed -i -e "s|%VERSION%|${dbox_version}|"       "${macos_dst_dir}/Info.plist"
+}
+
+pkg_msys2()
+{
+    install -DT "${build_dir}/dosbox.exe" "${pkg_dir}/dosbox.exe"
+
+    # Discover and copy required dll files
+    ntldd -R "${pkg_dir}/dosbox.exe" \
+        | sed -e 's/^[[:blank:]]*//g' \
+        | cut -d ' ' -f 3 \
+        | grep -E -i '(mingw|clang)(32|64)' \
+        | sed -e 's|\\|/|g' \
+        | xargs cp --target-directory="${pkg_dir}/"
+}
+
+pkg_msvc()
+{
+    # Get the release dir name from $build_dir
+    release_dir=$(basename "${build_dir}")
+
+    # Copy binary
+    cp "${build_dir}/dosbox.exe"  "${pkg_dir}/dosbox.exe"
+
+    # Copy dll files
+    cp "${build_dir}"/*.dll                  "${pkg_dir}/"
+    cp "src/libs/zmbv/${release_dir}"/*.dll  "${pkg_dir}/"
+
+    # Copy VC redistributable files
+    cp docs/vc_redist.txt                  "${pkg_dir}/doc/vc_redist.txt"
+    cp "$VC_REDIST_DIR/msvcp140.dll"       "${pkg_dir}/"
+    cp "$VC_REDIST_DIR/vcruntime140.dll"   "${pkg_dir}/"
+    cp "$VC_REDIST_DIR/vcruntime140_1.dll" "${pkg_dir}/" || true # might be missing, depending on arch
+}
+
+while getopts 'p:c:b:r:v:h' c
+do
+    case $c in
+        h) print_usage="true" ;;
+        p) platform=$OPTARG ;;
+        c) git_commit=$OPTARG ;;
+        b) git_branch=$OPTARG ;;
+        r) git_repo=$OPTARG ;;
+        v) dbox_version=$OPTARG ;;
+        *) true ;; # keep shellcheck happy
+    esac
+done
+
+shift "$((OPTIND - 1))"
+
+build_dir=$1
+pkg_dir=$2
+
+if [ "$print_usage" = "true" ]; then
+    usage
+    exit 0
+fi
+
+p=$platform
+case $p in
+    linux|macos|msys2|msvc) true ;;
+    *) platform="unsupported" ;;
+esac
+
+if [ "$platform" = "unsupported" ]; then
+    echo "Platform not set or unsupported"
+    usage
+    exit 1
+fi
+
+if [ ! -d "$build_dir" ]; then
+    echo "Build directory not set, or does not exist"
+    usage
+    exit 1
+fi
+
+if [ -z "$pkg_dir" ]; then
+    echo "Package directory not set"
+    usage
+    exit 1
+fi
```

Comment:
It would be nice to require that PACKAGE_DIR be either empty (or not exist).
At least, this is what we expect. 
This would also prevent clobbering existing files or accidentally "polluting" a directory that wasn't intended -- say, a user who sets the package_dir to their home folder (which would make a mess). 

If we later find some users want this behavior, then we add a `-f` (force) flag.

Reply:
Thought about it. Wasn't entirely sure if I should do so, but having a 'force' option sounds like a good compromise.